libbeat/outputs/redis/redis.go

```diff
@@ -119,18 +124,41 @@ func makeRedis(
 
 	clients := make([]outputs.NetworkClient, len(hosts))
 	for i, host := range hosts {
-		enc, err := codec.CreateEncoder(beat, config.Codec)
+		if parts := strings.SplitN(host, "://", 2); len(parts) != 2 {
+			host = redisScheme + host
+		}
+
+		hostUrl, err := url.Parse(host)
 		if err != nil {
 			return outputs.Fail(err)
 		}
 
+		if hostUrl.Scheme != redisScheme && hostUrl.Scheme != tlsRedisScheme {
+			return outputs.Fail(fmt.Errorf("invalid redis url scheme %s", hostUrl.Scheme))
+		}
+
+		if hostUrl.Scheme == tlsRedisScheme && transp.TLS == nil {
+			return outputs.Fail(errors.New("using 'rediss' scheme requires a tls config"))
+		}
```

Comment:
TLS can be nil. If "rediss" is given, we still want TLS support. The easiest might be create and empty instance here. The crypot/tls package do the rest for us.

Reply:
Ok